sdk/test/UnitTests/IPEndPointExceptionTests.cs

```diff
@@ -60,21 +60,55 @@ public class IPEndPointExceptionTests
             "127.12.1234.1:22"
         };
 
+        private readonly string[] _validDomains =
+        {
+            "example.com:0",
+            "example.com:65535",
+            "www.amazon.com:2000"
+        };
+
+        private readonly string[] _invalidDomains =
+        {
+            "example.com",
+            "example.com:70000",
+            "example.com:-100"
+        };
+
         [TestMethod]
         public void TestValidIpAddresses()
         {
             IPEndPoint endpoint;
-            foreach (var ip in _validInputs)
+            foreach (var ip in _validIPs)
             {
                 Assert.IsTrue(IPEndPointExtension.TryParse(ip, out endpoint));
             }
         }
 
+        [TestMethod]
+        public void TestValidDomains()

```

Comment:
Nice basic test. Lets add few tests here : https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/test/UnitTests/DaemonConfigTests.cs

The file covers all major tests for IPs, please add for domain name. We need to test `UpdateCache()`. Let me know if you can cover this in unit tests.

Reply:
Any update on this?

With hostname support lets include tests [here](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/test/UnitTests/DaemonConfigTests.cs).

The combination of valid inputs are : 
```
hostname:3001
tcp:hostname:3000 udp:127.0.0.1:3001
tcp:hostname:3000 udp:hostname:3001
```

Lets have `hostname` = `localhost` and in assertion check whether IP is `127.0.0.1`. This will validate resolving logic.

Reply:
Sure, `TestMultiThreadedUpdateSpacing` works.